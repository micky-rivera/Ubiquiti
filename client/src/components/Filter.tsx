import React, { useState } from 'react';
import { useAppDispatch, useAppSelector } from '../hooks/hooks';
import { setFilters } from '../slices/slices';

function Filter() {
    const state = useAppSelector(state => state.app);
    const dispatch = useAppDispatch();
    const [popUpOpen, setPopUpOpen] = useState(false);

    const handleClick = () => {
        if (popUpOpen) {
            setPopUpOpen(false);
        } else {
            setPopUpOpen(true);
        }
    }

    const handleCheck = (event: React.ChangeEvent<HTMLInputElement>) => {
        let storedFilters: string[] = [];
        if (window.localStorage.filters) {
            storedFilters = JSON.parse(window.localStorage.filters).terms;
        }
        if (event.target.checked === true) {
            if (!state.filters.includes(event.target.id)) {
                window.localStorage.setItem('filters', JSON.stringify({terms: [...storedFilters, event.target.id]}));
                dispatch(setFilters([...state.filters, event.target.id]));
            }
        } else {
            if (state.filters.includes(event.target.id)) {
                window.localStorage.setItem('filters', JSON.stringify({terms: [...storedFilters].filter(item => item !== event.target.id)}));
                dispatch(setFilters([...state.filters].filter(item => item !== event.target.id)))
            }
        }
    }

  return (
    <>
        <p onClick={handleClick} className="filter-btn">Filter</p>
        <div className={popUpOpen ? 'filter-popup' : 'hidden'}>
            <div className='filter-popup__header'>
                <p className='filter-popup__title'>Filter</p>
                <p onClick={handleClick} className='filter-popup__exit'>âœ•</p>
            </div>
            <div className='filter-popup__body'>
                <p className='filter-popup__section-title'>Product Line</p>
                <div className='filter-popup__option'>
                    <input className='filter-popup__checkbox' type='checkbox' id='UniFi' onChange={handleCheck} checked={state.filters.includes('UniFi')}/>
                    <label className='filter-popup__label' htmlFor='UniFi'>UniFi</label>
                </div>
                <div className='filter-popup__option'>
                    <input className='filter-popup__checkbox' type='checkbox' id='UniFi LTE' onChange={handleCheck} checked={state.filters.includes('UniFi LTE')}/>
                    <label className='filter-popup__label' htmlFor='UniFi LTE'>UniFi LTE</label>
                </div>
                <div className='filter-popup__option'>
                    <input className='filter-popup__checkbox' type='checkbox' id='UniFi LED' onChange={handleCheck} checked={state.filters.includes('UniFi LED')}/>
                    <label className='filter-popup__label' htmlFor='UniFi LED'>UniFi LED</label>
                </div>
                <div className='filter-popup__option'>
                    <input className='filter-popup__checkbox' type='checkbox' id='UniFi Protect' onChange={handleCheck} checked={state.filters.includes('UniFi Protect')}/>
                    <label className='filter-popup__label' htmlFor='UniFi Protect'>UniFi Protect</label>
                </div>
                <div className='filter-popup__option'>
                    <input className='filter-popup__checkbox' type='checkbox' id='UniFi Access' onChange={handleCheck} checked={state.filters.includes('UniFi Access')}/>
                    <label className='filter-popup__label' htmlFor='UniFi Access'>UniFi Access</label>
                </div>
                <div className='filter-popup__option'>
                    <input className='filter-popup__checkbox' type='checkbox' id='UniFi Talk' onChange={handleCheck} checked={state.filters.includes('UniFi Talk')}/>
                    <label className='filter-popup__label' htmlFor='UniFi Talk'>UniFi Talk</label>
                </div>
                <div className='filter-popup__option'>
                    <input className='filter-popup__checkbox' type='checkbox' id='AirMAX' onChange={handleCheck} checked={state.filters.includes('AirMAX')}/>
                    <label className='filter-popup__label' htmlFor='AirMAX'>AirMAX</label>
                </div>
                <div className='filter-popup__option'>
                    <input className='filter-popup__checkbox' type='checkbox' id='AirPrism' onChange={handleCheck} checked={state.filters.includes('AirPrism')}/>
                    <label className='filter-popup__label' htmlFor='AirPrism'>AirPrism</label>
                </div>
                <div className='filter-popup__option'>
                    <input className='filter-popup__checkbox' type='checkbox' id='EdgeMAX' onChange={handleCheck} checked={state.filters.includes('EdgeMAX')}/>
                    <label className='filter-popup__label' htmlFor='EdgeMAX'>EdgeMAX</label>
                </div>
                <div className='filter-popup__option'>
                    <input className='filter-popup__checkbox' type='checkbox' id='SunMAX' onChange={handleCheck} checked={state.filters.includes('SunMAX')}/>
                    <label className='filter-popup__label' htmlFor='SunMAX'>SunMAX</label>
                </div>
                <div className='filter-popup__option'>
                    <input className='filter-popup__checkbox' type='checkbox' id='AirFiber' onChange={handleCheck} checked={state.filters.includes('AirFiber')}/>
                    <label className='filter-popup__label' htmlFor='AirFiber'>AirFiber</label>
                </div>
                <div className='filter-popup__option'>
                    <input className='filter-popup__checkbox' type='checkbox' id='UFiber' onChange={handleCheck} checked={state.filters.includes('UFiber')}/>
                    <label className='filter-popup__label' htmlFor='UFiber'>UFiber</label>
                </div>
                <div className='filter-popup__option'>
                    <input className='filter-popup__checkbox' type='checkbox' id='LTU' onChange={handleCheck} checked={state.filters.includes('LTU')}/>
                    <label className='filter-popup__label' htmlFor='LTU'>LTU</label>
                </div>
                <div className='filter-popup__option'>
                    <input className='filter-popup__checkbox' type='checkbox' id='AmpliFi' onChange={handleCheck} checked={state.filters.includes('AmpliFi')}/>
                    <label className='filter-popup__label' htmlFor='AmpliFi'>AmpliFi</label>
                </div>
                <div className='filter-popup__option'>
                    <input className='filter-popup__checkbox' type='checkbox' id='mFi' onChange={handleCheck} checked={state.filters.includes('mFi')}/>
                    <label className='filter-popup__label' htmlFor='mFi'>mFi</label>
                </div>
                <div className='filter-popup__option'>
                    <input className='filter-popup__checkbox' type='checkbox' id='UISP' onChange={handleCheck} checked={state.filters.includes('UISP')}/>
                    <label className='filter-popup__label' htmlFor='UISP'>UISP</label>
                </div>
                <div className='filter-popup__option'>
                    <input className='filter-popup__checkbox' type='checkbox' id='UNMS' onChange={handleCheck} checked={state.filters.includes('UNMS')}/>
                    <label className='filter-popup__label' htmlFor='UISP'>UNMS</label>
                </div>
                <div className='filter-popup__option'>
                    <input className='filter-popup__checkbox' type='checkbox' id='Smart' onChange={handleCheck} checked={state.filters.includes('Smart')}/>
                    <label className='filter-popup__label' htmlFor='Smart'>Smart</label>
                </div>
                <div className='filter-popup__option'>
                    <input className='filter-popup__checkbox' type='checkbox' id='SmartPower' onChange={handleCheck} checked={state.filters.includes('SmartPower')}/>
                    <label className='filter-popup__label' htmlFor='SmartPower'>SmartPower</label>
                </div>
                <div className='filter-popup__option'>
                    <input className='filter-popup__checkbox' type='checkbox' id='Security' onChange={handleCheck} checked={state.filters.includes('Security')}/>
                    <label className='filter-popup__label' htmlFor='Security'>Security</label>
                </div>
                <div className='filter-popup__option'>
                    <input className='filter-popup__checkbox' type='checkbox' id='Conference' onChange={handleCheck} checked={state.filters.includes('Conference')}/>
                    <label className='filter-popup__label' htmlFor='Conference'>Conference</label>
                </div>
                <div className='filter-popup__option'>
                    <input className='filter-popup__checkbox' type='checkbox' id='Ubiquiti' onChange={handleCheck} checked={state.filters.includes('Ubiquiti')}/>
                    <label className='filter-popup__label' htmlFor='Ubiquiti'>Ubiquiti</label>
                </div>
            </div>
        </div>
    </>
  );
}

export default Filter;
