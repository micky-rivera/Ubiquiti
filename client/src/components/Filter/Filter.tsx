import React, { useState } from 'react';
import { useAppDispatch, useAppSelector } from '../../hooks/hooks';
import { setFilters } from '../../slices/slices';

function Filter() {
    const state = useAppSelector(state => state.app);
    const dispatch = useAppDispatch();
    const [popUpOpen, setPopUpOpen] = useState(false);

    const handleClick = () => {
        if (popUpOpen) {
            setPopUpOpen(false);
        } else {
            setPopUpOpen(true);
        }
    }

    const handleCheck = (event: React.ChangeEvent<HTMLInputElement>) => {
        let storedFilters: string[] = [];
        if (window.localStorage.filters) {
            storedFilters = JSON.parse(window.localStorage.filters).terms;
        }
        if (event.target.checked === true) {
            if (!state.filters.includes(event.target.id.toLowerCase())) {
                window.localStorage.setItem('filters', JSON.stringify({terms: [...storedFilters, event.target.id.toLowerCase()]}));
                dispatch(setFilters([...state.filters, event.target.id.toLowerCase()]));
            }
        } else {
            if (state.filters.includes(event.target.id.toLowerCase())) {
                window.localStorage.setItem('filters', JSON.stringify({terms: [...storedFilters].filter(item => item !== event.target.id.toLowerCase())}));
                dispatch(setFilters([...state.filters].filter(item => item !== event.target.id.toLowerCase())));
            }
        }
    }

  return (
    <>
        <p onClick={handleClick} className="filter-btn" data-testid='filter-btn'>Filter</p>
        <div className={popUpOpen ? 'filter-popup' : 'hidden'}>
            <div className='filter-popup__header'>
                <p className='filter-popup__title'>Filter</p>
                <p onClick={handleClick} className='filter-popup__exit'>âœ•</p>
            </div>
            <div className='filter-popup__body' data-testid='filter-popup__body'>
                <p className='filter-popup__section-title'>Product Line</p>
                <div className='filter-popup__option'>
                    <input className='filter-popup__checkbox' type='checkbox' id='UniFi' onChange={handleCheck} checked={state.filters.includes('unifi')}/>
                    <label className='filter-popup__label' htmlFor='UniFi'>UniFi</label>
                </div>
                <div className='filter-popup__option'>
                    <input className='filter-popup__checkbox' type='checkbox' id='UniFi LTE' onChange={handleCheck} checked={state.filters.includes('unifi lte')}/>
                    <label className='filter-popup__label' htmlFor='UniFi LTE'>UniFi LTE</label>
                </div>
                <div className='filter-popup__option'>
                    <input className='filter-popup__checkbox' type='checkbox' id='UniFi LED' onChange={handleCheck} checked={state.filters.includes('unifi led')}/>
                    <label className='filter-popup__label' htmlFor='UniFi LED'>UniFi LED</label>
                </div>
                <div className='filter-popup__option'>
                    <input className='filter-popup__checkbox' type='checkbox' id='UniFi Protect' onChange={handleCheck} checked={state.filters.includes('unifi protect')}/>
                    <label className='filter-popup__label' htmlFor='UniFi Protect'>UniFi Protect</label>
                </div>
                <div className='filter-popup__option'>
                    <input className='filter-popup__checkbox' type='checkbox' id='UniFi Access' onChange={handleCheck} checked={state.filters.includes('unifi access')}/>
                    <label className='filter-popup__label' htmlFor='UniFi Access'>UniFi Access</label>
                </div>
                <div className='filter-popup__option'>
                    <input className='filter-popup__checkbox' type='checkbox' id='UniFi Talk' onChange={handleCheck} checked={state.filters.includes('unifi talk')}/>
                    <label className='filter-popup__label' htmlFor='UniFi Talk'>UniFi Talk</label>
                </div>
                <div className='filter-popup__option'>
                    <input className='filter-popup__checkbox' type='checkbox' id='AirMAX' onChange={handleCheck} checked={state.filters.includes('airmax')}/>
                    <label className='filter-popup__label' htmlFor='AirMAX'>AirMAX</label>
                </div>
                <div className='filter-popup__option'>
                    <input className='filter-popup__checkbox' type='checkbox' id='AirPrism' onChange={handleCheck} checked={state.filters.includes('airprism')}/>
                    <label className='filter-popup__label' htmlFor='AirPrism'>AirPrism</label>
                </div>
                <div className='filter-popup__option'>
                    <input className='filter-popup__checkbox' type='checkbox' id='AirCube' onChange={handleCheck} checked={state.filters.includes('aircube')}/>
                    <label className='filter-popup__label' htmlFor='AirCube'>AirCube</label>
                </div>
                <div className='filter-popup__option'>
                    <input className='filter-popup__checkbox' type='checkbox' id='EdgeMAX' onChange={handleCheck} checked={state.filters.includes('edgemax')}/>
                    <label className='filter-popup__label' htmlFor='EdgeMAX'>EdgeMAX</label>
                </div>
                <div className='filter-popup__option'>
                    <input className='filter-popup__checkbox' type='checkbox' id='SunMAX' onChange={handleCheck} checked={state.filters.includes('sunmax')}/>
                    <label className='filter-popup__label' htmlFor='SunMAX'>SunMAX</label>
                </div>
                <div className='filter-popup__option'>
                    <input className='filter-popup__checkbox' type='checkbox' id='AirFiber' onChange={handleCheck} checked={state.filters.includes('airfiber')}/>
                    <label className='filter-popup__label' htmlFor='AirFiber'>AirFiber</label>
                </div>
                <div className='filter-popup__option'>
                    <input className='filter-popup__checkbox' type='checkbox' id='UFiber' onChange={handleCheck} checked={state.filters.includes('ufiber')}/>
                    <label className='filter-popup__label' htmlFor='UFiber'>UFiber</label>
                </div>
                <div className='filter-popup__option'>
                    <input className='filter-popup__checkbox' type='checkbox' id='LTU' onChange={handleCheck} checked={state.filters.includes('ltu')}/>
                    <label className='filter-popup__label' htmlFor='LTU'>LTU</label>
                </div>
                <div className='filter-popup__option'>
                    <input className='filter-popup__checkbox' type='checkbox' id='AmpliFi' onChange={handleCheck} checked={state.filters.includes('amplifi')}/>
                    <label className='filter-popup__label' htmlFor='AmpliFi'>AmpliFi</label>
                </div>
                <div className='filter-popup__option'>
                    <input className='filter-popup__checkbox' type='checkbox' id='mFi' onChange={handleCheck} checked={state.filters.includes('mfi')}/>
                    <label className='filter-popup__label' htmlFor='mFi'>mFi</label>
                </div>
                <div className='filter-popup__option'>
                    <input className='filter-popup__checkbox' type='checkbox' id='UISP' onChange={handleCheck} checked={state.filters.includes('uisp')}/>
                    <label className='filter-popup__label' htmlFor='UISP'>UISP</label>
                </div>
                <div className='filter-popup__option'>
                    <input className='filter-popup__checkbox' type='checkbox' id='UNMS' onChange={handleCheck} checked={state.filters.includes('unms')}/>
                    <label className='filter-popup__label' htmlFor='UISP'>UNMS</label>
                </div>
                <div className='filter-popup__option'>
                    <input className='filter-popup__checkbox' type='checkbox' id='Smart' onChange={handleCheck} checked={state.filters.includes('smart')}/>
                    <label className='filter-popup__label' htmlFor='Smart'>Smart</label>
                </div>
                <div className='filter-popup__option'>
                    <input className='filter-popup__checkbox' type='checkbox' id='SmartPower' onChange={handleCheck} checked={state.filters.includes('smartpower')}/>
                    <label className='filter-popup__label' htmlFor='SmartPower'>SmartPower</label>
                </div>
                <div className='filter-popup__option'>
                    <input className='filter-popup__checkbox' type='checkbox' id='Security' onChange={handleCheck} checked={state.filters.includes('security')}/>
                    <label className='filter-popup__label' htmlFor='Security'>Security</label>
                </div>
                <div className='filter-popup__option'>
                    <input className='filter-popup__checkbox' type='checkbox' id='Conference' onChange={handleCheck} checked={state.filters.includes('conference')}/>
                    <label className='filter-popup__label' htmlFor='Conference'>Conference</label>
                </div>
                <div className='filter-popup__option'>
                    <input className='filter-popup__checkbox' type='checkbox' id='Ubiquiti' onChange={handleCheck} checked={state.filters.includes('ubiquiti')}/>
                    <label className='filter-popup__label' htmlFor='Ubiquiti'>Ubiquiti</label>
                </div>
            </div>
        </div>
    </>
  );
}

export default Filter;
