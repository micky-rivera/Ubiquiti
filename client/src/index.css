@import './styles/header.css';
@import './styles/list.css';
@import './styles/nav.css';
@import './styles/grid.css';

* {
  margin: 0px;
  padding: 0px;
  border: 0px;
  box-sizing: border-box;
}

.hidden {
  display: none;
}

body {
  margin: 0;
  font-family: 'Lato', sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

code {
  font-family: source-code-pro, Menlo, Monaco, Consolas, 'Courier New',
    monospace;
}
