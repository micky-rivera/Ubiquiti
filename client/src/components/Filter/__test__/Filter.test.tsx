import React from "react";
import Filter from '../Filter';
import { Provider } from 'react-redux'
import {store} from '../../../store/store'
import { render, cleanup } from "@testing-library/react";
import TestRenderer from 'react-test-renderer';
import { createRoot } from 'react-dom/client';

afterEach(cleanup);

it('renders without crashing', () => {
    const div = document.createElement('div');
    const root = createRoot(div);
    root.render(<Provider store={store}><Filter /></Provider>)
})

it('renders the button with the word "filter', () => {
    const { getByTestId } = render(<Provider store={store}><Filter /></Provider>);
    expect(getByTestId('filter-btn')).toHaveTextContent('Filter');
})

it('renders the body of the popup', () => {
    const { getByTestId } = render(<Provider store={store}><Filter /></Provider>);
    expect(getByTestId('filter-popup__body')).toHaveTextContent('Product Line');
})

it('renders the very last check item', () => {
    const { getByTestId } = render(<Provider store={store}><Filter /></Provider>);
    expect(getByTestId('filter-popup__body')).toHaveTextContent('Ubiquiti');
})

it('matches snapshot', () => {
    const tree = TestRenderer.create(<Provider store={store}><Filter /></Provider>).toJSON();
    expect(tree).toMatchSnapshot();
})