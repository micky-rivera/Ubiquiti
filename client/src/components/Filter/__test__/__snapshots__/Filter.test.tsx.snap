// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
Array [
  <p
    className="filter-btn"
    data-testid="filter-btn"
    onClick={[Function]}
  >
    Filter
  </p>,
  <div
    className="hidden"
  >
    <div
      className="filter-popup__header"
    >
      <p
        className="filter-popup__title"
      >
        Filter
      </p>
      <p
        className="filter-popup__exit"
        onClick={[Function]}
      >
        âœ•
      </p>
    </div>
    <div
      className="filter-popup__body"
      data-testid="filter-popup__body"
    >
      <p
        className="filter-popup__section-title"
      >
        Product Line
      </p>
      <div
        className="filter-popup__option"
      >
        <input
          checked={false}
          className="filter-popup__checkbox"
          id="UniFi"
          onChange={[Function]}
          type="checkbox"
        />
        <label
          className="filter-popup__label"
          htmlFor="UniFi"
        >
          UniFi
        </label>
      </div>
      <div
        className="filter-popup__option"
      >
        <input
          checked={false}
          className="filter-popup__checkbox"
          id="UniFi LTE"
          onChange={[Function]}
          type="checkbox"
        />
        <label
          className="filter-popup__label"
          htmlFor="UniFi LTE"
        >
          UniFi LTE
        </label>
      </div>
      <div
        className="filter-popup__option"
      >
        <input
          checked={false}
          className="filter-popup__checkbox"
          id="UniFi LED"
          onChange={[Function]}
          type="checkbox"
        />
        <label
          className="filter-popup__label"
          htmlFor="UniFi LED"
        >
          UniFi LED
        </label>
      </div>
      <div
        className="filter-popup__option"
      >
        <input
          checked={false}
          className="filter-popup__checkbox"
          id="UniFi Protect"
          onChange={[Function]}
          type="checkbox"
        />
        <label
          className="filter-popup__label"
          htmlFor="UniFi Protect"
        >
          UniFi Protect
        </label>
      </div>
      <div
        className="filter-popup__option"
      >
        <input
          checked={false}
          className="filter-popup__checkbox"
          id="UniFi Access"
          onChange={[Function]}
          type="checkbox"
        />
        <label
          className="filter-popup__label"
          htmlFor="UniFi Access"
        >
          UniFi Access
        </label>
      </div>
      <div
        className="filter-popup__option"
      >
        <input
          checked={false}
          className="filter-popup__checkbox"
          id="UniFi Talk"
          onChange={[Function]}
          type="checkbox"
        />
        <label
          className="filter-popup__label"
          htmlFor="UniFi Talk"
        >
          UniFi Talk
        </label>
      </div>
      <div
        className="filter-popup__option"
      >
        <input
          checked={false}
          className="filter-popup__checkbox"
          id="AirMAX"
          onChange={[Function]}
          type="checkbox"
        />
        <label
          className="filter-popup__label"
          htmlFor="AirMAX"
        >
          AirMAX
        </label>
      </div>
      <div
        className="filter-popup__option"
      >
        <input
          checked={false}
          className="filter-popup__checkbox"
          id="AirPrism"
          onChange={[Function]}
          type="checkbox"
        />
        <label
          className="filter-popup__label"
          htmlFor="AirPrism"
        >
          AirPrism
        </label>
      </div>
      <div
        className="filter-popup__option"
      >
        <input
          checked={false}
          className="filter-popup__checkbox"
          id="AirCube"
          onChange={[Function]}
          type="checkbox"
        />
        <label
          className="filter-popup__label"
          htmlFor="AirCube"
        >
          AirCube
        </label>
      </div>
      <div
        className="filter-popup__option"
      >
        <input
          checked={false}
          className="filter-popup__checkbox"
          id="EdgeMAX"
          onChange={[Function]}
          type="checkbox"
        />
        <label
          className="filter-popup__label"
          htmlFor="EdgeMAX"
        >
          EdgeMAX
        </label>
      </div>
      <div
        className="filter-popup__option"
      >
        <input
          checked={false}
          className="filter-popup__checkbox"
          id="SunMAX"
          onChange={[Function]}
          type="checkbox"
        />
        <label
          className="filter-popup__label"
          htmlFor="SunMAX"
        >
          SunMAX
        </label>
      </div>
      <div
        className="filter-popup__option"
      >
        <input
          checked={false}
          className="filter-popup__checkbox"
          id="AirFiber"
          onChange={[Function]}
          type="checkbox"
        />
        <label
          className="filter-popup__label"
          htmlFor="AirFiber"
        >
          AirFiber
        </label>
      </div>
      <div
        className="filter-popup__option"
      >
        <input
          checked={false}
          className="filter-popup__checkbox"
          id="UFiber"
          onChange={[Function]}
          type="checkbox"
        />
        <label
          className="filter-popup__label"
          htmlFor="UFiber"
        >
          UFiber
        </label>
      </div>
      <div
        className="filter-popup__option"
      >
        <input
          checked={false}
          className="filter-popup__checkbox"
          id="LTU"
          onChange={[Function]}
          type="checkbox"
        />
        <label
          className="filter-popup__label"
          htmlFor="LTU"
        >
          LTU
        </label>
      </div>
      <div
        className="filter-popup__option"
      >
        <input
          checked={false}
          className="filter-popup__checkbox"
          id="AmpliFi"
          onChange={[Function]}
          type="checkbox"
        />
        <label
          className="filter-popup__label"
          htmlFor="AmpliFi"
        >
          AmpliFi
        </label>
      </div>
      <div
        className="filter-popup__option"
      >
        <input
          checked={false}
          className="filter-popup__checkbox"
          id="mFi"
          onChange={[Function]}
          type="checkbox"
        />
        <label
          className="filter-popup__label"
          htmlFor="mFi"
        >
          mFi
        </label>
      </div>
      <div
        className="filter-popup__option"
      >
        <input
          checked={false}
          className="filter-popup__checkbox"
          id="UISP"
          onChange={[Function]}
          type="checkbox"
        />
        <label
          className="filter-popup__label"
          htmlFor="UISP"
        >
          UISP
        </label>
      </div>
      <div
        className="filter-popup__option"
      >
        <input
          checked={false}
          className="filter-popup__checkbox"
          id="UNMS"
          onChange={[Function]}
          type="checkbox"
        />
        <label
          className="filter-popup__label"
          htmlFor="UISP"
        >
          UNMS
        </label>
      </div>
      <div
        className="filter-popup__option"
      >
        <input
          checked={false}
          className="filter-popup__checkbox"
          id="Smart"
          onChange={[Function]}
          type="checkbox"
        />
        <label
          className="filter-popup__label"
          htmlFor="Smart"
        >
          Smart
        </label>
      </div>
      <div
        className="filter-popup__option"
      >
        <input
          checked={false}
          className="filter-popup__checkbox"
          id="SmartPower"
          onChange={[Function]}
          type="checkbox"
        />
        <label
          className="filter-popup__label"
          htmlFor="SmartPower"
        >
          SmartPower
        </label>
      </div>
      <div
        className="filter-popup__option"
      >
        <input
          checked={false}
          className="filter-popup__checkbox"
          id="Security"
          onChange={[Function]}
          type="checkbox"
        />
        <label
          className="filter-popup__label"
          htmlFor="Security"
        >
          Security
        </label>
      </div>
      <div
        className="filter-popup__option"
      >
        <input
          checked={false}
          className="filter-popup__checkbox"
          id="Conference"
          onChange={[Function]}
          type="checkbox"
        />
        <label
          className="filter-popup__label"
          htmlFor="Conference"
        >
          Conference
        </label>
      </div>
      <div
        className="filter-popup__option"
      >
        <input
          checked={false}
          className="filter-popup__checkbox"
          id="Ubiquiti"
          onChange={[Function]}
          type="checkbox"
        />
        <label
          className="filter-popup__label"
          htmlFor="Ubiquiti"
        >
          Ubiquiti
        </label>
      </div>
    </div>
  </div>,
]
`;
